### ISSUE TITLE
Issues with TeXstudio 4.4.0 (that were not present in 4.3.1)

### ISSUE THREAD

**Issue by HolgerGerhardt** [texstudio-org/texstudio#2691]
### Environment
- TeXstudio: 4.4.0
- Qt: 6.4.0
- OS: macOS 13.0.1
- TeX distribution: MacTeX-2022

### Expected behavior

1. TeXstudio should tell macOS that it can open _.tex_ files.
2. Everything that I type should be displayed immediately.

### Actual behaviour
1. macOS is unaware that TeXstudio 4.4.0 can open _.tex_ files: <img width="646" alt="image" src="https://user-images.githubusercontent.com/45849777/202896969-ec557c30-005e-487d-993d-a6a3c239f8e6.png">
2. With TeXstudio 4.4.0, as soon as the document is rather long, TeX commands like `\usepackage`, `\textbf`, etc. are only displayed with a delay of an entire second (on an M1 Pro MacBook Pro). This does not happen in short documents, and it did not happen in 4.3.1.

### How to reproduce

1. When picking “Open With” from the context menu in Finder, TeXstudio 4.4.0 is not listed even though it is present (and open). Earlier versions of TeXstudio are listed.
2. Compare version 4.4.0 with 4.3.1 on macOS: Typing TeX commands in 4.3.1 (with Qt v6.3.1) is noticeably quicker in long documents than in 4.4.0. This is even the case when I turn off auto-completion of TeX commands. Hence, this seems to be an issue with the check for valid TeX commands being too slow. This issue is so severe that it makes 4.4.0 unusable for me, and I have downgraded to 4.3.1.

**Comment by sunderme** [comment]
can you try https://github.com/texstudio-org/texstudio/suites/9406371441/artifacts/444053244 (.tex association)

**Comment by HolgerGerhardt** [comment]
> can you try https://github.com/texstudio-org/texstudio/suites/9406371441/artifacts/444053244 (.tex association)

Yes, that solves the missing association with the _.tex_ filename extension for me. Thank you!

**Comment by sunderme** [comment]
the other issue I will not be able to debug as I don't have a mac available.

**Comment by HolgerGerhardt** [comment]
An additional observation regarding

> ### Actual behaviour
> 2. With TeXstudio 4.4.0, as soon as the document is rather long, TeX commands like `\usepackage`, `\textbf`, etc. are only displayed with a delay of an entire second (on an M1 Pro MacBook Pro). This does not happen in short documents, and it did not happen in 4.3.1.

The delay also occurs in the following situation: Say, I have a file _main.tex_ which includes `\input{introduction.tex}` (and several other `\input`s). In _main.tex_, autocompletion is (unbearably) slow. (That file is also loaded slowly, even though it is rather small, but it consists of various `\input`s.)

* When I open _introduction.tex_ alone, autocompletion is as fast as I would it expect to be.
* As soon as I haven both _main.tex_ and _introduction.tex_ open simultaneously, autocompletion is also (unbearably) slow in _introduction.tex_.
* When I close _main.tex_, autocompletion still remains slow in _introduction.tex_.
* Only after I closing and reopening _introduction.tex_, autocompletion is quick again.

**Comment by sunderme** [comment]
that sounds like a problem with your project set-up
Can you send it to me to check ? (sunderme@gmx.de)

**Comment by cpierquet** [comment]
Hi-, for me (W11 ans TXS4.4.0) autocompletion is sometimes a bit slow when editing existing document, as shown in the screencast provided ;-)

https://user-images.githubusercontent.com/45103962/202916060-a8ab8146-063c-4a0f-b3ea-5369502a8756.mp4

**Comment by sunderme** [comment]
yes, here because it is usepackage with an open argumen (open brace, no closing brace)
txs tries to interpret all following as a cwl package name. I will probably need to implement an early fail in such cases but that is probably something different.

**Comment by FloMiLe** [comment]
I can confirm that typing basically any command gets delayed substantially in the Mac version. As soon as the backslash key is hit, all further input gets displayed about a second later.

Is there a way I could help with fixing this bug?

**Comment by sunderme** [comment]
1) screencast to demonstrate the issue.
It is related to the completer being open ?
2) if you can, compile txs locally and see if the issue persists.
Local compilation without pdf viewer is easy (for testing). Getting the pdf viewer on mac is a bit of a hassle.
3) if you know to use a debugger, you can try to pinpoint the issue in the code.

**Comment by mevang** [comment]
\documentclass[]{article}
\begin{document}

\textbf{P}

\end{document}

For some reason on my machine  \textbf{P} doesnt work .  Every thing else works.

**Comment by FloMiLe** [comment]
@sunderme, sorry for the delay. 

1. Yes, it is related to the completer being activated or not. When the completer is active, it seems that delays are caused by the completer getting ready. Delays seem to only happen directly after opening a file that is part of a bigger project, but also after closing the Preferences window. Please see my screen recording attached.

2., 3. I, unfortunately, do not know at all how to compile or debug an app. – I'm sorry.

But I have read a few times now that your not having a Mac prevents you from testing texstudio on macOS. So, in all seriousness, could I maybe contribute to your great project by sending you an older MacBook Pro that I don't use anymore? If so, can you suggest a way to exchange contact details privately without posting them publicly?

While I was preparing (and practicing) for the screen recording, I noticed (and demonstrate in the video) some minor behaviors of the editor that maybe you could consider implementing:
1. An empty completer should vanish immediately, not require e.g. Enter.
2. An open completer should not inhibit "Delete left word".
3. "Delete left word" should consider consecutive macros as separate words.
(All deletions to the left in the video were done using the shortcut for "Delete left word".)

https://user-images.githubusercontent.com/13892710/203797266-c268332e-a4fb-4375-8f24-86c0c35cb523.mp4

**Comment by sunderme** [comment]
you can contact me via sunderme@gmx.de

**Comment by sunderme** [comment]
the delay when opening the completer the first time after changing options or changing includes is normal (though not nice)
I probably have to find a better solution to prepare the command database. Right now it is prepared that first time, hence the delay.

**Comment by FloMiLe** [comment]
I have compared loading times of main files of bigger projects and of the completer using the two latest releases. In both cases, version 4.4.0 took about 40 to 60% longer than version 4.3.1, e.g. 14s versus 6s for loading a main file.

**Comment by sunderme** [comment]
without concrete test case, that is difficult to examine.
You could send me the document to my email.
I don't think that this is an osx issue.

**Comment by helour** [comment]
Hello, some icons are missing in version 4.4.0. These icons were in version 4.3.1 but are not in the latest version.
![MissingIcons](https://user-images.githubusercontent.com/79715809/204095143-180e7a78-d8a1-44dd-b8ab-2be7da04812c.png)
Please fix it if possible.
Thx

**Comment by sunderme** [comment]
let's close this.
I don't multiple bugs in one issue as it impedes keeping track what is resolved and what not.
For the perceived loss of speed, I need a test case. Open a new issue with a test case (or send it directly to me).

**Comment by HolgerGerhardt** [comment]
> For the perceived loss of speed, I need a test case. Open a new issue with a test case (or send it directly to me).

Sorry for not having been able to reply earlier!

I have opened a new issue: https://github.com/texstudio-org/texstudio/issues/2823.
